xarray/backends/h5netcdf_.py

```diff
@@ -353,6 +353,8 @@ def close(self, **kwargs):
 
 class H5netcdfBackendEntrypoint(BackendEntrypoint):
     available = has_h5netcdf
+    description = "Open netCDF files (.nc, .nc4 and .cdf) using h5netcdf in Xarray"
+    url = "https://docs.xarray.dev/en/stable/generated/xarray.backends.H5netcdfBackendEntrypoint.html"
```

Comment:
Interesting idea. I wonder whether these should be in docstrings then be linked by Sphinx, vs. in code. But ofc fine to try from my perspective.

Reply:
Just one comment here, `h5netcdf` (and `netcdf4`) can also open hdf5 files which are not strict NetCDF4. Not sure how to better phrase this, though.

Reply:
> Interesting idea. I wonder whether these should be in docstrings then be linked by Sphinx, vs. in code. But ofc fine to try from my perspective.

No idea how that would work, could you give an example?

Reply:
> Just one comment here, `h5netcdf` (and `netcdf4`) can also open hdf5 files which are not strict NetCDF4. Not sure how to better phrase this, though.

Should ".h5" be added to `guess_can_open`?
Or you don't want to specify this because not strict netCDF files might not work with xarray?

Reply:
> Should ".h5" be added to `guess_can_open`?

I'd say that this is not necessary. Those who want to open will use the engine-kwarg.

There are only a few hdf5-specifics which can't be read by `netcdf4` (see https://h5netcdf.org/#invalid-netcdf-files, list might not be up-to-date as netcdf-c is evolving too). So xarray is able digest most _normal_ hdf5-files (via `netcdf4` and `h5netcdf`).

My concern is, that users might conclude that only netCDF4-files are readable and get confused. Good chance that my impression will not hold and users know what they are doing.

Anyway, I really like these enhancements to the backends.